src/mds/CInode.cc

```diff
@@ -1949,6 +1949,13 @@ void CInode::finish_scatter_update(ScatterLock *lock, CDir *dir,
 	pf->rstat.version = pi->rstat.version;
 	pf->accounted_rstat = pf->rstat;
 	ename = "lock inest accounted scatter stat update";
+
+	if (!is_auth() && lock->get_next_state() == LOCK_MIX) {
+	  dout(10) << "finish_scatter_update try to assimilate dirty rstat on " 
+	    << *dir << dendl; 
+	  dir->assimilate_dirty_rstat_inodes();
+       }
+
```

Comment:
we need to move "pf->version = dir->pre_dirty()" after assimilate_dirty_rstat_inodes. because MutationImpl::apply() first pop dirty inodes, then pop dirty fnodes. pf->version should be assigned at last. I think this is why assertion in CDir::commit is triggered.